### ISSUE TITLE
Sometimes sent mail remove subjects

### ISSUE THREAD

**Issue by jancborchardt** [owncloud/mail#949]
Sometimes when I reply to mails, the subject gets removed for some reason. Can you reproduce @irgendwie @ChristophWurst @PoPoutdoor @zinks-?

**Comment by DeepDiver1975** [comment]
> Can you reproduce

no

**Comment by irgendwie** [comment]
Also happens for me sometimes :+1: therefore, can reproduce

**Comment by PoPoutdoor** [comment]
Can not reproduce subject lost, but quoted contents get stripped to retain first line and the second line is [...]

**Comment by jancborchardt** [comment]
@irgendwie can you maybe dig deeper into this? :)

**Comment by jancborchardt** [comment]
Just had this happen again on a reply … it’s really embarassing. How can I debug?

**Comment by DeepDiver1975** [comment]
Just observed this as well .... mail app really feels buggy and broken these days .... I wonder if we should fork it or start off in a rework branch .....

**Comment by jancborchardt** [comment]
LOL :D

**Comment by jancborchardt** [comment]
We need to write more tests, and hold back on adding more features. ;)

**Comment by PoPoutdoor** [comment]
> I wonder if we should fork it or start off in a rework branch .....

When you all settled with #944, a rework branch seems the logical path.

**Comment by jancborchardt** [comment]
@PoPoutdoor I’m pretty sure this was a joke by @DeepDiver1975 … rework branches have brought us nothing but despair in the past. ;) We need to make small steps and incremental improvements instead of trying to work on one big thing which never gets finished.

**Comment by PoPoutdoor** [comment]
@jancborchardt I do feel something broken... btw, is there any flow chart or diagram of Mail app?

**Comment by ChristophWurst** [comment]
'Sometimes' = if you reply from the unified inbox ;-)

The problem are those lines: https://github.com/owncloud/mail/blob/master/js/views/composer.js#L320-L341 as the 'folderId' is not set when replying from the unified inbox. This results in a request without the 'folderId' parameter. Furthermore the server interprets this as a new message because it checks for 'accountId' and 'folderId' existence here: https://github.com/owncloud/mail/blob/master/lib/controller/accountscontroller.php#L289-L305

Summing up, instead of a reply, the server sends a new message. Since we don't send the original subject to the server when replying (it's extracted from the original message), the message sent has no subject.
